modules/Sources/Features/Receive/ReceiveStore.swift

```diff
@@ -0,0 +1,102 @@
+//
+//  ReceiveStore.swift
+//  
+//
+//  Created by Matthew Watt on 7/15/23.
+//
+
+import ComposableArchitecture
+import Generated
+import Pasteboard
+import SDKSynchronizer
+import ZcashLightClientKit
+
+public typealias ReceiveStore = Store<ReceiveReducer.State, ReceiveReducer.Action>
+public typealias ReceiveViewStore = ViewStore<ReceiveReducer.State, ReceiveReducer.Action>
+
+public struct ReceiveReducer: ReducerProtocol {
+    public struct State: Equatable {
+        public enum Toast {
+            case copiedToClipboard
+        }
+        
+        @BindingState public var toast: Toast?
+        
+        public var unifiedAddress: String {
+            uAddress?.stringEncoded ?? L10n.AddressDetails.Error.cantExtractUnifiedAddress
+        }
+        
+        public var transparentAddress: String {
+            do {
+                let address = try uAddress?.transparentReceiver().stringEncoded ?? L10n.AddressDetails.Error.cantExtractTransparentAddress
+                return address
+            } catch {
+                return L10n.AddressDetails.Error.cantExtractTransparentAddress
+            }
+        }
+
+        public var saplingAddress: String {
+            do {
+                let address = try uAddress?.saplingReceiver().stringEncoded ?? L10n.AddressDetails.Error.cantExtractSaplingAddress
+                return address
+            } catch {
+                return L10n.AddressDetails.Error.cantExtractSaplingAddress
+            }
+        }
+        
+        var uAddress: UnifiedAddress?
+        
+        public init(uAddress: UnifiedAddress? = nil) {
+            self.uAddress = uAddress
+        }
+    }
+    
+    public enum Action: BindableAction, Equatable {
+        case binding(BindingAction<State>)
+        case onAppear
+        case uAddressChanged(UnifiedAddress?)
+        case showQrCodeTapped
+        case copyPrivateAddressTapped
+        case topUpWalletTapped
+        case copyTransparentAddressTapped
+    }
+    
+    @Dependency(\.pasteboard) var pasteboard
+    @Dependency(\.sdkSynchronizer) var sdkSynchronizer
+    
+    public init () {}
+    
+    public var body: some ReducerProtocol<State, Action> {
+        BindingReducer()
+        
+        Reduce { state, action in
+            switch action {
+            case .onAppear:
+                return .task {
+                    return .uAddressChanged(try? await sdkSynchronizer.getUnifiedAddress(0))
+                }
+
+            case .uAddressChanged(let uAddress):
+                state.uAddress = uAddress
+                return .none
+            case .copyPrivateAddressTapped:
```

Comment:
[non-blockling]

this action's name is deceiving because a UA may contain transparent receiver as well.

Reply:
oh, probably Github is not pointing out the correct line. I'm specifically referring to `copyPrivateAddressTapped`